internal/blockchain/common/common.go

```diff
@@ -177,48 +188,74 @@ func (cb *callbacks) BatchPinOrNetworkAction(ctx context.Context, subInfo *Subsc
 			batch.TransactionType = core.TransactionTypeContractInvokePin
 		}
 	}
-	return cb.batchPinComplete(ctx, []string{subInfo.V2Namespace}, batch, signingKey)
+	cb.addBatchPinComplete(ctx, events, []string{subInfo.V2Namespace}, batch, signingKey)
 }
 
-func (cb *callbacks) batchPinComplete(ctx context.Context, namespaces []string, batch *blockchain.BatchPin, signingKey *core.VerifierRef) error {
+func (cb *callbacks) addBatchPinComplete(ctx context.Context, events EventsToDispatch, namespaces []string, batch *blockchain.BatchPin, signingKey *core.VerifierRef) {
 	for _, namespace := range namespaces {
-		if handler, ok := cb.handlers[namespace]; ok {
-			if err := handler.BatchPinComplete(namespace, batch, signingKey); err != nil {
-				return err
-			}
+		if _, ok := cb.handlers[namespace]; ok {
+			events[namespace] = append(events[namespace], &blockchain.EventToDispatch{
```

Comment:
Do we need locks around this and the other places we append to the map+list? Maybe this will only be required when we make each namespace have its own event listener?

Reply:
Great point - checking 👍